docs/guide/troubleshooting.md

```diff
@@ -41,4 +41,56 @@ kubectl port-forward -n "${APPLICATION_NAMESPACE}" \
+ 
+You can use the below command to figure out the exact reason of the envoy bootup issue. If the error is due to not being able to mount SDS provider's UDS socket then you would need to address that.
+
+```bash
+kubectl describe pod <pod-name> -n <namespace-name> 
+```
+
+### Pod is up and running but Envoy doesn’t have any certs in SDS mode.
+
+1. To begin with, check if `APPMESH_SDS_SOCKET_PATH` env variable is present under Envoy and if it has the correct UDS path value. If it is missing, then the controller didn’t inject the env variable. Check if `enable-sds` is set to `true` for the controller.
+
+For example, when using SPIRE Agent you should see something like below in Envoy.
+
+```
+APPMESH_SDS_SOCKET_PATH:      /run/spire/sockets/agent.sock
+```
+
+2. If the above env variable is present with the correct value, then check if Envoy is able to communicate with the SDS Provider. Below command will help verify if the Envoy is able to reach out to the local SDS Provider via the UDS path passed in to the controller and also to see if it is healthy.
+
+```bash
+kubectl exec -it <pod-name> -n <namespace-name> -c envoy -- curl http://localhost:9901/clusters | grep -E '(static_cluster_sds.*cx_active|static_cluster_sds.*healthy)'
+
+static_cluster_sds_unix_socket::/run/spire/sockets/agent.sock::cx_active::1
+static_cluster_sds_unix_socket::/run/spire/sockets/agent.sock::health_flags::healthy
+```
+
+3. If the SDS cluster is healthy in Envoy, then check if the workload entry tied to this particular Pod/app container is registered with SPIRE Server and if the selectors match. Use the below command to list out all the registered workload entries.
+
+```bash
+kubectl exec -n spire spire-server-0 -- /opt/spire/bin/spire-server entry show
+```
+Once you have the list of entries, check for the entry that is tied to the app container and check if the selectors match. Default selectors that we currently use are pod’s service account, namespace and labels.
+
+### Pod liveness and readiness probes fail when mTLS is enabled
+
+HTTP and TCP health checks from the kubelet will not work as is, if mutual TLS is enabled as the kubelet doesn't have relevant certs.
+
+**Workarounds:**
+
+1. Expose the health check endpoint on a different port and skip mTLS for that port. You can then set `appmesh.k8s.aws/ports` annotation with the application port value on the deployment spec.
+
+**Example**: If your main application port is 8080 and if health check endpoint is exposed on 8081, then `appmesh.k8s.aws/ports:8080` will help bypass mTLS for health check port.
```

Comment:
Can we add some helpful commands here ? For instance how to find health check endpoint port ?

Reply:
health check end point port that we are referring to here is what the customer picks to expose the health check of their application.